Release date

alpgarcia opened the pull request:
Adds `release_date` field to panels. Also adds `strict` argument for uploading panels iif they are newer than the ones in `.kibana`.

Added a second commit to manage exceptions in order to show `content` part of responses, where ElasticSearch puts the actual reason of the error.

To use this approach with ElasticSearch + Kibana 6 we would need to modify `.kibana` mapping to include the new field because index mapping is currently set to strict, which means that no dynamic fields are allowe. The right way to do it is:
```
PUT .kibana/_mapping/doc 
{
  "properties": {
    "dashboard": {
      "properties": {
        "release_date": {
          "type": "date"
        }
      }
    }
  }
}
```
After that, the code should be working for ES + Kibana 6 too.

Commit:
Add release_date to panel json

This change adds a new field `release_date` to panels exported using kidash. It also
adds a new `strict` argument that allows to upload a panel only when its release
date is newer than the one stored in `.kibana`.

Commit:
Add try/except to show content of error responses

ElasticSearch returns error details within content. This change captures exceptions
from `kidash` binary and uses logger to print them.

Commit:
Add constant for release date field name

acs commented:
After thinking about it, and in order to simplify the use of kibana and the compatibility between Elasticsearch versions the best approach is to modify the mapping if needed inside of kidash. Could you do it @alpgarcia ? Thanks!

Commit:
Update mapping in Kibana6 if needed

When using Kibana 6, .kibana index mapping is set to strict. That means we cannot
include new fields in the index without explicitely add the corresponding mapping
for that field. This change detects when this error is raised, updates .kibana
mapping including `release_date`  field and tries to upload panel again.

alpgarcia commented:
This change automatically updates Kibana 6 `.kibana` mapping to support `release_date` field when error raises.

I tried several cases:
* Import json file with no release date:
  * kidash strict mode: raises an error as it must be present.
  * kidash normal mode: updates the panel as usual. If `release_date` exists in the panel that is going to be overwritten, that field will be lost.
* Import json file with release date:
  * kidash strict mode: it only updates the panel if it is newer than the current one, based on `release_date`
  * kidash normal mode:  panel will be overwritten with the new one, regardless `release_date` field values.

acs reviewed (approved):
LGTM
